app/qml/MerginProjectPanel.qml

```diff
@@ -27,730 +27,769 @@ Item {
     height: width
-    running: false
+    running: stackView.pending
     visible: running
     anchors.centerIn: parent
-    z: authPanel.z + 1
+    z: parent.z + 1
   }
 
-  PanelHeader {
-    id: header
-    height: InputStyle.rowHeightHeader
-    width: parent.width
-    color: InputStyle.clrPanelMain
-    rowHeight: InputStyle.rowHeightHeader
-    titleText: qsTr("Projects")
-
-    onBack: {
-      if (authPanel.visible) {
-        authPanel.visible = false
-        homeBtn.activated()
-      } else {
-        projectsPanel.visible = false
+  Component {
+    id: projectsPanelComp
+    Item {
+
+      objectName: "projectsPanel"
+
+      function getStatusIcon(status, pending) {
+        if (pending) return "stop.svg"
+
+        if (status === "noVersion") return "download.svg"
+        else if (status === "outOfDate") return "sync.svg"
+        else if (status === "upToDate") return "check.svg"
+        else if (status === "modified") return "sync.svg"
+
+        return "more_menu.svg"
       }
-    }
-    withBackButton: projectsPanel.activeProjectPath || authPanel.visible
 
-    Item {
-      id: avatar
-      width: InputStyle.rowHeightHeader * 0.8
-      height: InputStyle.rowHeightHeader
-      anchors.right: parent.right
-      anchors.rightMargin: projectsPanel.panelMargin
+      function refreshProjectList() {
+        if (toolbar.highlighted === exploreBtn.text) {
+          exploreBtn.activated()
+        } else if (toolbar.highlighted === sharedProjectsBtn.text) {
+          sharedProjectsBtn.activated()
+        } else if (toolbar.highlighted === myProjectsBtn.text) {
+          myProjectsBtn.activated()
+        } else homeBtn.activated()
+      }
 
-      Rectangle {
-        id: avatarImage
-        anchors.centerIn: parent
-        width: avatar.width
-        height: avatar.width
-        color: InputStyle.fontColor
-        radius: width*0.5
-        antialiasing: true
+      Component.onCompleted: {
+        // load model just after all components are prepared
+        // otherwise GridView's delegate item is initialized invalidately
+        grid.model = __projectsModel
+        merginProjectsList.model = __merginProjectsModel
+      }
 
-        MouseArea {
-          anchors.fill: parent
-          onClicked: {
-            if (__merginApi.userAuth.hasAuthData() && __merginApi.apiVersionStatus === MerginApiStatus.OK) {
-              __merginApi.getUserInfo()
-              accountPanel.visible = true
-              reloadList.visible = false
+      Connections {
+        target: __projectsModel
+        onModelReset: {
+           var index = __projectsModel.rowAccordingPath(activeProjectPath)
+           if (index !== activeProjectIndex) {
+            activeProjectIndex = index
+           }
+        }
+      }
+
+      Connections {
+        target: __merginApi
+        onListProjectsFinished: {
+          stackView.pending = false
+        }
+        onListProjectsFailed: {
+          reloadList.visible = true
```

Comment:
Do we still need to set visibility of pages even with stackview?

Reply:
Its a button to manually refresh project list if it failed before.